ekncontent/ekncontent/eknc-dictionary-object-model.c

```diff
@@ -0,0 +1,199 @@
+
+G_DEFINE_TYPE_WITH_PRIVATE (EkncDictionaryObjectModel,
+                            eknc_dictionary_object_model,
+                            EKNC_TYPE_CONTENT_OBJECT_MODEL)
+
+enum {
+  PROP_0,
+  PROP_WORD,
+  PROP_DEFINITION,
+  PROP_PART_OF_SPEECH,
+  NPROPS
+};
+
+static GParamSpec *eknc_dictionary_object_model_props [NPROPS] = { NULL, };
+
+static void
+eknc_dictionary_object_model_get_property (GObject    *object,
+                                           guint       prop_id,
+                                           GValue     *value,
+                                           GParamSpec *pspec)
+{
+  EkncDictionaryObjectModel *self = EKNC_DICTIONARY_OBJECT_MODEL (object);
+  EkncDictionaryObjectModelPrivate *priv = eknc_dictionary_object_model_get_instance_private (self);
+
+  switch (prop_id)
+    {
+    case PROP_WORD:
+      g_value_set_string (value, priv->word);
+      break;
+
+    case PROP_DEFINITION:
+      g_value_set_string (value, priv->definition);
+      break;
+
+    case PROP_PART_OF_SPEECH:
+      g_value_set_string (value, priv->part_of_speech);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+eknc_dictionary_object_model_set_property (GObject *object,
+                                           guint prop_id,
+                                           const GValue *value,
+                                           GParamSpec *pspec)
+{
+  EkncDictionaryObjectModel *self = EKNC_DICTIONARY_OBJECT_MODEL (object);
+  EkncDictionaryObjectModelPrivate *priv = eknc_dictionary_object_model_get_instance_private (self);
+
+  switch (prop_id)
+    {
+    case PROP_WORD:
+      priv->word = g_value_dup_string (value);
+      break;
+
+    case PROP_DEFINITION:
+      priv->definition = g_value_dup_string (value);
+      break;
+
+    case PROP_PART_OF_SPEECH:
+      priv->part_of_speech = g_value_dup_string (value);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+eknc_dictionary_object_model_finalize (GObject *object)
+{
+  EkncDictionaryObjectModel *self = EKNC_DICTIONARY_OBJECT_MODEL (object);
+  EkncDictionaryObjectModelPrivate *priv = eknc_dictionary_object_model_get_instance_private (self);
+
+  g_clear_pointer (&priv->word, g_free);
+  g_clear_pointer (&priv->definition, g_free);
+  g_clear_pointer (&priv->part_of_speech, g_free);
+
+  G_OBJECT_CLASS (eknc_dictionary_object_model_parent_class)->finalize (object);
+}
+
+static void
+eknc_dictionary_object_model_class_init (EkncDictionaryObjectModelClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->get_property = eknc_dictionary_object_model_get_property;
+  object_class->set_property = eknc_dictionary_object_model_set_property;
+  object_class->finalize = eknc_dictionary_object_model_finalize;
+
+  /**
+   * EkncDictionaryObjectModel:word:
+   *
+   * The actual word of the day
```

Comment:
These documentation comments and property descriptions should make sense in the broader context of the eos-knowledge-content documentation, and not be so specific to word-of-the-day